enumerable_method.rb

```diff
@@ -29,180 +30,252 @@ def my_select
-        t = if_is_a_class(x, param)
-        break if t == false
+        return false unless x
       end
     end
     t
   end
 
   def my_any?(param = nil)
-    return false if param.class == Regexp
     return false if empty?
 
     t = true
     if block_given?
-      return true unless block_given?
-
       my_each do |x|
-        t = if_nil_false_return_false yield(x)
+        t = yield(x).nil? || yield(x) == false ? false : true
         break if t == true
       end
-    elsif param.nil?
-      my_each do |x|
-        t = if_nil_false_return_false x
-        break if t == true
+    elsif !param.nil?
+      if param.class == Class
+        my_each do |x|
+          t = x.is_a?(param) ? true : false
+          break if t == true
+        end
+      elsif param.class == Regexp
+        my_each do |x|
+          t = x.match(param) ? true : false
+          break if t == true
+        end
+      else
+        my_each do |x|
+          return true unless x == param
+        end
       end
     else
       my_each do |x|
-        t = if_is_a_class(x, param)
+        t = x == false || x.nil? ? false : true
       end
     end
     t
   end
 
   def my_none?(param = nil)
-    return true if param.class == Regexp
-
     t = true
     if block_given?
-      return true unless block_given?
-
       my_each do |x|
-        t = if_false_return_true(yield(x))
+        t = yield(x) == false
         break if t == false
       end
-    elsif param.nil?
+    elsif param.class == Class && !param.nil?
       my_each do |x|
-        t = if_true_return_false(x)
+        t = x.is_a?(param) ? false : true
         break if t == false
       end
-    else
+    elsif param.class == Regexp && !param.nil?
       my_each do |x|
-        t = if_is_a_class(x, param)
+        t = x.match(param) ? false : true
+        break if t == true
+      end
+    else
+      t = false
+      return true if length.zero? && param.nil?
+
+      if param.nil?
+        my_each do |x|
+          t = x != true
+          break if t == false
+        end
+      else
+        my_each do |x|
+          t = x != param
+          break if t == false
+        end
       end
     end
     t
   end
+  # rubocop:enable Metrics/CyclomaticComplexity
+  # rubocop:enable Metrics/MethodLength
+  # rubocop:enable Metrics/PerceivedComplexity
 
-  def my_count(item = nil)
-    counter = 0
-    if item.nil?
-      return length unless block_given?
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/PerceivedComplexity
 
-      each do |x|
-        counter += 1 if yield(x)
+  def my_inject(acum_initial = nil, symb = nil)
+    if block_given?
+      if is_a? Array
+        return 0 if length.zero? && acum_initial.nil?
+        return nil if length.zero?
+
+        acum = acum_initial.nil? ? self[0] : acum_initial
+        self[0..length].my_each { |item| acum = yield(item, acum) }
```

Comment:
Let's handle this case too.
For cases where `[1,2,3]` is the enumerable, for example, `1` is already assigned to `acum` however, it is still included in the array to be counted again (`self[0..length]`), making it to be evaluated two times. See example:

```ruby
irb(main):016:0> [1,2,3].inject{|sum, b| sum + b}
=> 6
irb(main):017:0> [1,2,3].my_inject{|sum, b| sum + b}
=> 7
```

The above example shows how `1` is been added two times instead of one.

Reply:
than you, I fix your recommendation